### PR TITLE
[BUGFIX] Use 'public' as webDirectory

### PR DESCRIPTION
**Pull request by simonschaufi**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
Bugfix

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Since TYPO3 9 the "official" web directory is 'public'. This is also default in many other frameworks like Laravel or Symfony.

* **What is the current behavior?** (You can also link to an open issue here)
web root directory is 'web'

* **What is the new behavior (if this is a feature change)?**
web root directory is 'public'

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Users who rely on surf defaults must change the option "webDirectory" to 'web'

### CODE REVIEW

File: src/Task/Php/WebOpcacheResetCreateScriptTask.php [hunk 1]
```diff
@@ -92,7 +92,7 @@ public function execute(Node $node, Application $application, Deployment $deploy
         $options = $this->configureOptions($options);
 
         $workspacePath = $deployment->getWorkspacePath($application);
-        $webDirectory = $application->hasOption('webDirectory') ? $application->getOption('webDirectory') : 'Web';
+        $webDirectory = $application->hasOption('webDirectory') ? $application->getOption('webDirectory') : 'public';
```
Comment (t3easy) [inline src/Task/Php/WebOpcacheResetCreateScriptTask.php]: Could we set the default based on application?

Comment (simonschaufi) [reply src/Task/Php/WebOpcacheResetCreateScriptTask.php]: I added all the defaults in the different applications

Comment (simonschaufi) [reply src/Task/Php/WebOpcacheResetCreateScriptTask.php]: As you can see before it was totally messy. Sometimes "web" was used, sometimes "Web"

Comment (t3easy) [reply src/Task/Php/WebOpcacheResetCreateScriptTask.php]: > As you can see before it was totally messy. Sometimes "web" was used, sometimes "Web"

It was no mess. The "Web" comes from the Flow. At this time, TYPO3 had no web folder. Then TYPO3 introduced "web" as default and now changed to "public" as default.

Comment (simonschaufi) [reply src/Task/Php/WebOpcacheResetCreateScriptTask.php]: so will you approve or do I need to change something?

Comment (t3easy) [reply src/Task/Php/WebOpcacheResetCreateScriptTask.php]: I would add a constant for the default public folder to each app and use it as fallback. Somethink like 
```
$webDirectory = $application->hasOption('webDirectory') ? $application->getOption('webDirectory') : $application::defaultPublicFolder;
```

Comment (simonschaufi) [reply src/Task/Php/WebOpcacheResetCreateScriptTask.php]: Is it ok to have the global Application default also set to 'public' or not?

File: src/Domain/Model/Application.php [hunk 2]
```diff
@@ -22,6 +22,8 @@ class Application
      */
     public const DEFAULT_SHARED_DIR = 'shared';
 
+    public const DEFAULT_WEB_DIRECTORY = 'public';
```
Comment (simonschaufi) [inline src/Domain/Model/Application.php]: das ist der globale Default. Könnte man auch noch anpassen auf `''`

File: src/Task/Php/WebOpcacheResetCreateScriptTask.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Task/Php/WebOpcacheResetCreateScriptTask.php]: @simonschaufi You should also set a defaultPublicFolder constant for the default Application. Yes.

**Pull request merged**